apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,34 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgClass, NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import { Component, EventEmitter, Input, Output, TemplateRef } from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   template: `
     <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4" [class]="customClass">
+        <ng-content select="[cardImage]"></ng-content>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track $index) {
+        <app-list-item [id]="item.id" (delete)="delete.emit($event)">
+          <ng-container *ngTemplateOutlet="listTemplateRef || emptyTemplateRef; context:{$implicit:item}"></ng-container> 
+        </app-list-item>
+        }
       </section>
-
-      <button
-        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
-        Add
-      </button>
+      <ng-content select="[cardActionButton]"></ng-content>
     </div>
+    <ng-template #emptyTemplateRef>
+      <p>No List Template Found.</p>
+    </ng-template>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet, NgClass],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
-
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
+export class CardComponent<T extends {id:number}> {
+  @Input() list: T[] | null = null;
+  @Input() customClass !:string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @Input() listTemplateRef: TemplateRef<any> | null = null;
 
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @Output() delete: EventEmitter<number> = new EventEmitter();
```

Comment:
you can use the new set of signal queries

Reply:
Thank you for your suggestion. I appreciate your input and will certainly explore the possibility of replacing this LOC  with signal queries. At the moment, I'm not fully acquainted with this approach, but I'm committed to learning and implementing best practices.